### PR TITLE
8344263: GenShen: Reduce extraneous log messages at INFO level

### PR DESCRIPTION
**Pull request by earthling-amzn**
Genshen logs a lot of fine level detail at `gc=info`. This category should be reserved for major phase changes in the gc cycle.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed (1 review required, with at least 1 [Committer](https://openjdk.org/bylaws#committer))

### Issue
 * [JDK-8344263](https://bugs.openjdk.org/browse/JDK-8344263): GenShen: Reduce extraneous log messages at INFO level (**Task** - P4)

### Reviewers
 * [Y. Srinivas Ramakrishna](https://openjdk.org/census#ysr) (@ysramakrishna - Committer) ⚠️ Review applies to [f02d0245](https://git.openjdk.org/shenandoah/pull/540/files/f02d02457a47c6a54ad91136b45a6694d36ca173)
 * [Aleksey Shipilev](https://openjdk.org/census#shade) (@shipilev - Committer)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/shenandoah.git pull/540/head:pull/540` \
`$ git checkout pull/540`

Update a local copy of the PR: \
`$ git checkout pull/540` \
`$ git pull https://git.openjdk.org/shenandoah.git pull/540/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 540`

View PR using the GUI difftool: \
`$ git pr show -t 540`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/shenandoah/pull/540.diff">https://git.openjdk.org/shenandoah/pull/540.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/shenandoah/pull/540#issuecomment-2477771356)
</details>

### CODE REVIEW

**Review by ysramakrishna — APPROVED** [review]
Can you share an example snippet (in the ticket or in the PR) of what we would see with a young and an old cycle with the reduced info-level logging?

The changes to convert from info to debug all look good to me.

**Comment by earthling-amzn** [comment]
@shipilev - meet in the middle? Log heuristic's decisions at `info` with tag `ergo`, the rest of the changes in this PR continue logging at `debug`.

@ysramakrishna - A typical young cycle looks like this:
````
[3.055s][info][gc] Trigger (YOUNG): Average GC time (22.07 ms) is above the time for average allocation rate (53621 KB/s) to deplete free headroom (0B) (margin of error = 1.91)
[3.055s][info][gc] GC(29) Concurrent reset (Young) 0.042ms
[3.055s][info][gc] GC(29) Pause Init Mark (Young) 0.037ms
[3.055s][info][gc] GC(29) Concurrent remembered set scanning 0.064ms
[3.056s][info][gc] GC(29) Concurrent marking roots 0.278ms
[3.057s][info][gc] GC(29) Concurrent marking (Young) 1.779ms
[3.058s][info][gc] GC(29) Pause Final Mark (Young) 0.116ms
[3.058s][info][gc] GC(29) Concurrent weak references (Young) 0.017ms
[3.058s][info][gc] GC(29) Concurrent weak roots (Young) 0.116ms
[3.058s][info][gc] GC(29) Concurrent cleanup (Young) 30M->30M(32M) 0.003ms
```
A typical cycle for the non-generational mode looks like this:
```
[2.729s][info][gc] Trigger (GLOBAL): Average GC time (52.17 ms) is above the time for average allocation rate (20698 KB/s) to deplete free headroom (0B) (margin of error = 3.29)
[2.729s][info][gc] GC(42) Concurrent reset  (unload classes) 0.068ms
[2.730s][info][gc] GC(42) Pause Init Mark  (unload classes) 0.018ms
[2.730s][info][gc] GC(42) Concurrent marking roots 0.390ms
[2.735s][info][gc] GC(42) Concurrent marking  (unload classes) 4.471ms
[2.735s][info][gc] GC(42) Pause Final Mark  (unload classes) 0.111ms
[2.735s][info][gc] GC(42) Concurrent thread roots 0.080ms
[2.735s][info][gc] GC(42) Concurrent weak references  (unload classes) 0.047ms
[2.736s][info][gc] GC(42) Concurrent weak roots  (unload classes) 0.740ms
[2.736s][info][gc] GC(42) Concurrent cleanup  (unload classes) 4M->4M(32M) 0.002ms
[2.744s][info][gc] GC(42) Concurrent class unloading 8.163ms
[2.744s][info][gc] GC(42) Concurrent strong roots 0.158ms
[2.745s][info][gc] GC(42) Concurrent evacuation 0.420ms
[2.745s][info][gc] GC(42) Pause Init Update Refs 0.008ms
[2.747s][info][gc] GC(42) Concurrent update references 2.494ms
[2.747s][info][gc] GC(42) Concurrent update thread roots 0.310ms
[2.748s][info][gc] GC(42) Pause Final Update Refs 0.044ms
[2.748s][info][gc] GC(42) Concurrent cleanup  (unload classes) 13M->12M(32M) 0.029ms
```
The `Trigger` still shows the generational name in non-generational modes. I'll see about that.

File: src/hotspot/share/gc/shenandoah/shenandoahUtils.hpp [hunk 1]
```diff
@@ -47,16 +47,16 @@ class ShenandoahGeneration;
 #define SHENANDOAH_RETURN_EVENT_MESSAGE(generation_type, prefix, postfix) \
   switch (generation_type) {                                              \
     case NON_GEN:                                                         \
-      return prefix " (NON-GENERATIONAL)" postfix;                        \
+      return prefix " " postfix;                                          \
```
Comment (shipilev) [inline src/hotspot/share/gc/shenandoah/shenandoahUtils.hpp:50]: I think this causes a double space, for example here between "roots" and "(unload classes)" here:

```
[2.736s][info][gc] GC(42) Concurrent weak roots  (unload classes) 0.740ms
```

**Comment by shipilev** [comment]
> @shipilev - meet in the middle? Log heuristic's decisions at `info` with tag `ergo`, the rest of the changes in this PR continue logging at `debug`.

Sure, OK.

**Comment by earthling-amzn** [comment]
@ysramakrishna - I've changed the generation names to also be proper case for consistency with other usages. On second thought, I don't think we need the generation name in the trigger messages. It's pretty clear from context what's being collected (non-generational mode):
```
[2.644s][info][gc] Trigger: Average GC time (52.87 ms) is above the time for average allocation rate (20071 KB/s) to deplete free headroom (0B) (margin of error = 3.27)
[2.644s][info][gc] GC(40) Concurrent reset (unload classes) 0.077ms
[2.645s][info][gc] GC(40) Pause Init Mark (unload classes) 0.057ms
```
(generational mode):
```
[3.176s][info][gc] Trigger: Average GC time (14.74 ms) is above the time for average allocation rate (51928 KB/s) to deplete free headroom (542K) (margin of error = 2.01)
[3.176s][info][gc] GC(28) Concurrent reset (Young) 0.075ms
[3.176s][info][gc] GC(28) Pause Init Mark (Young) 0.074ms
```

File: src/hotspot/share/gc/shenandoah/shenandoahUtils.hpp (hunk 1 above)
Comment (earthling-amzn) [reply src/hotspot/share/gc/shenandoah/shenandoahUtils.hpp]: Yes, fixed here: https://github.com/openjdk/shenandoah/pull/540/commits/1c7752c121853c131a352025f3d87cc179549506

**Comment by ysramakrishna** [comment]
One question re:

> [3.055s][info][gc] Trigger: Average GC time (22.07 ms) is above the time for average allocation rate (53621 KB/s) to deplete free headroom (0B) (margin of error = 1.91)
[3.055s][info][gc] GC(29) Concurrent reset (Young) 0.042ms
[3.055s][info][gc] GC(29) Pause Init Mark (Young) 0.037ms
[3.055s][info][gc] GC(29) Concurrent remembered set scanning 0.064ms
[3.056s][info][gc] GC(29) Concurrent marking roots 0.278ms
[3.057s][info][gc] GC(29) Concurrent marking (Young) 1.779ms
[3.058s][info][gc] GC(29) Pause Final Mark (Young) 0.116ms
[3.058s][info][gc] GC(29) Concurrent weak references (Young) 0.017ms
[3.058s][info][gc] GC(29) Concurrent weak roots (Young) 0.116ms
[3.058s][info][gc] GC(29) Concurrent cleanup (Young) 30M->30M(32M) 0.003ms

Is there a reason not to use "(Young)" against the remainder of the phases as well for clarity, when appropriate? Also wondering if the same would be true of the Trigger.

I imagine there are Triggers for Old. How are they logged?

**Review by shipilev — APPROVED** [review]
I looked at `-Xlog:gc` with this patch, and it looks good. I say we do this.

**Pull request closed without merging**